src/pmap_getport.c

```diff
@@ -71,22 +71,29 @@ u_short pmap_getport(struct sockaddr_in *address, u_long program,
 			       RPCSMALLMSGSIZE, RPCSMALLMSGSIZE);
 	if (client != NULL) {
 		struct clnt_req *cc = mem_alloc(sizeof(*cc));
+		enum clnt_stat stat;
 
 		parms.pm_prog = program;
 		parms.pm_vers = version;
 		parms.pm_prot = protocol;
 		parms.pm_port = 0;	/* not needed or used */
 
-		clnt_req_fill(cc, client, authnone_create(), PMAPPROC_GETPORT,
+		clnt_req_fill(cc, client, authnone_ncreate(), PMAPPROC_GETPORT,
 			      (xdrproc_t) xdr_pmap, &parms,
 			      (xdrproc_t) xdr_u_short, &port);
-		if (clnt_req_setup(cc, to)) {
-			if (CLNT_CALL_WAIT(cc) != RPC_SUCCESS) {
-				rpc_createerr.cf_stat = RPC_PMAPFAILURE;
-				clnt_geterr(client, &rpc_createerr.cf_error);
-			} else if (port == 0) {
-				rpc_createerr.cf_stat = RPC_PROGNOTREGISTERED;
-			}
+		stat = clnt_req_setup(cc, to);
+		if (stat == RPC_SUCCESS) {
+			stat = CLNT_CALL_WAIT(cc);
+		}
+		if (stat != RPC_SUCCESS) {
+			char *t = rpc_sperror(&cc->cc_error, __func__);
```

Comment:
This section seems to pop up a lot.  Maybe an inline function or macro?  I suspect a free will be missed at some point.